ctapipe/reco/hillas_reconstructor.py

```diff
@@ -307,99 +231,107 @@ def initialize_hillas_planes(
         mode and the correction to the psi angle is explained in [gasparetto]_
         section 7.1.4.
         """
+        # get one telescope id to check what frame to use
+        altaz = AltAz()
 
-        hillas_planes = {}
-
-        # dictionary to store the telescope-wise image directions
-        # to be projected on the ground and corrected in case of mispointing
-        corrected_angle_dict = {}
+        # Due to tracking the pointing of the array will never be a constant
+        array_pointing = SkyCoord(
+            az=event.pointing.array_azimuth,
+            alt=event.pointing.array_altitude,
+            frame=altaz,
+        )
 
-        k = next(iter(telescopes_pointings))
-        horizon_frame = telescopes_pointings[k].frame
-        for tel_id, moments in hillas_dict.items():
+        # create arrays / quantities of all needed things for all telescopes
+        # so we can do transformations vectorized
+        cog1 = np.empty(len(hillas_dict))
+        cog2 = np.empty(len(hillas_dict))
+        cam_radius = np.empty(len(hillas_dict))
+        psi = np.empty(len(hillas_dict))
+        weights = np.empty(len(hillas_dict))
+        focal_length = np.empty(len(hillas_dict))
+        alt = np.empty(len(hillas_dict))
+        az = np.empty(len(hillas_dict))
+        tel_ids = np.empty(len(hillas_dict), dtype=int)
+
+        hillas_in_camera_frame = False
+
+        for i, (tel_id, hillas) in enumerate(hillas_dict.items()):
+            tel_ids[i] = tel_id
+
+            pointing = event.pointing.tel[tel_id]
+            alt[i] = pointing.altitude.to_value(u.rad)
+            az[i] = pointing.azimuth.to_value(u.rad)
+
+            if isinstance(hillas, CameraHillasParametersContainer):
+                hillas_in_camera_frame = True
```

Comment:
`hillas_in_camera_frame = isinstance(hillas, CameraHillasParametersContainer)`?

Reply:
Doesn't really save anything, since the if block contains more statements. Moving it outside the loop is not possible

Reply:
I misread the indentation level of `hillas_in_camera_frame` in L296.